test/unit-test/ice_api_private/ut.cmake

```diff
@@ -0,0 +1,87 @@
+# Include filepaths for source and include.
+include( ${MODULE_ROOT_DIR}/iceFilePaths.cmake )
+include( ${MODULE_ROOT_DIR}/source/dependency/amazon-kinesis-video-streams-stun/stunFilePaths.cmake )
+
+# ====================  Define your project name (edit) ========================
+set( project_name "ice_api_private" )
+
+message( STATUS "${project_name}" )
+
+# =====================  Create your mock here  (edit)  ========================
+
+# List the files to mock here.
+list(APPEND mock_list
+            "${MODULE_ROOT_DIR}/source/include/ice_data_types.h"
+        )
+# List the directories your mocks need.
+list(APPEND mock_include_list
+            ${ICE_INCLUDE_PUBLIC_DIRS}
+            ${MODULE_ROOT_DIR}/test/unit-test
+        )
+
+# List the definitions of your mocks to control what to be included.
+list(APPEND mock_define_list
+            ""
+       )
+
+# ================= Create the library under test here (edit) ==================
+
+# List the files you would like to test here.
+list(APPEND real_source_files
+            ${MODULE_ROOT_DIR}/source/ice_api_private.c
+            ${MODULE_ROOT_DIR}/source/ice_api.c
+            ${MODULE_ROOT_DIR}/source/transaction_id_store.c
+            ${STUN_SOURCES}
```

Comment:
Is it possible to make STUN an external sources? Then we can control it by CMOCK?

Reply:
Yes, based on the requirement. If it felt necessary later to Mock STUN. We will do these changes.